rasa/core/validator.py

```diff
@@ -57,13 +58,42 @@ def verify_intents(self, ignore_warnings: bool = True) -> bool:
 
         return everything_is_alright
 
+    def verify_example_repetition_in_intents(
+        self, ignore_warnings: bool = True
+    ) -> bool:
+        """Checks if there is no duplicated example in different intents."""
+
+        everything_is_alright = True
+
+        duplication_hash = defaultdict(set)
+        duplicated_examples = {}
+        for example in self.intents.intent_examples:
+            text = example.text
+            duplication_hash[text].add(example.get("intent"))
+
+            if len(duplication_hash[text]) > 1:
+                duplicated_examples[text] = duplication_hash[text]
+                everything_is_alright = ignore_warnings and everything_is_alright
+
+        for text, intents in duplicated_examples.items():
```

Comment:
why do we need additional `duplicated_examples`? can we iterate over `duplication_hash` here?

Reply:
Yes, we can. I can see two forms to do so, but in both of them there is a problem associated.

Reply:
Yes, we can. I created the variable `duplicated_examples` to keep the complexity of the code between O(2n) and O(n + m) where m < n and m is the number of keys at `duplicated_examples`.
If we iterate over `duplication_hash` again after the first iteration that fills its keys and values, we will also need to check again if the length of the set for a key text is bigger than 1 to show the warning to the user. This second iteration over `duplication_hash` would also make the complexity of the code be O(2n) since `duplication_hash` holds all the examples of the nlu file.

Reply:
The code without the `duplicated_examples` would be something like this
```
everything_is_alright = True

duplication_hash = defaultdict(set)
for example in self.intents.intent_examples:
    text = example.text
    duplication_hash[text].add(example.get("intent"))

for text, intents in duplication_hash.items():
    if len(duplication_hash[text]) > 1:
        everything_is_alright = ignore_warnings and everything_is_alright
        logger.warning(
            "The example '{}' was found in these multiples intents: {}".format(
                text, ", ".join(sorted(intents))
            )
        )
return everything_is_alright
```

Reply:
@Ghostvv, which solution do you think is better to keep? The one that uses `duplicated_examples` or the one that iterates over `duplication_hash`?

Reply:
I made the change. The code above works fine and don't break the tests.